### PR TITLE
fix lazy-loading by checking if we're already past the 'VimEnter' event

### LINKED ISSUE #49: Commands aren't registered properly when lazy loading with packer.nvim
The only lazy-loading mechanism that works for nvim-ide is `event = 'VimEnter'` which sorta defeats the purpose of lazy-loading.

Here's my current config:

```lua
return {
  -- localplugin just returns the local path to the cloned repo
  -- if it exists on disk
  localplugin('ldelossa/nvim-ide'),
  event = 'VimEnter',
  config = function()
    local explorer = require('ide.components.explorer')
    local outline = require('ide.components.outline')
    local bufferlist = require('ide.components.bufferlist')
    local changes = require('ide.components.changes')
    local commits = require('ide.components.commits')
    require('ide').setup({
      components = {
        [explorer.Name] = {
          list_directories_first = true,
          show_file_permissions = false,
          keymaps = {
            hide = '<NOP>',
          },
        },
        [outline.Name] = {
          keymaps = {
            hide = '<NOP>',
          },
        },
      },
      panel_groups = {
        explorer = { bufferlist.Name, explorer.Name, outline.Name },
        git = { changes.Name, commits.Name },
      },
      workspaces = {
        auto_close = false,
      },
    })
  end,
}
```

However if I change `event = 'VimEnter'` to `cmd = 'Workspace'`, the command doesn't get registered properly and I'm unable to open the panels.

### PR DESCRIPTION
**Pull request by mrjones2014**
Fixes #49 

It was checking `VimEnter`, but if we're lazy loading the plugin, the plugin will never see that event because it's already fired by the time the plugin is loaded.

### CODE REVIEW

**Comment by mrjones2014** [comment]
This is fixed now.

**Comment by ldelossa** [comment]
The VimEnter dance is necessary when *not* lazy loading, or else we dont wind up restoring the cursor to the empty buffer thats created when vim starts. 

My concern is that this may break for non-lazy loaders. Is this OK?

**Comment by mrjones2014** [comment]
This won't break for non lazy-loaders because we check `vim.v.vim_did_enter` -- if false, we do exactly what you were doing before, if true, we just initialize right away because `VimEnter` already happened.

File: .luacheckrc [hunk 1]
```diff
@@ -0,0 +1 @@
+globals = { "vim" }
```
[the remark this answers is not in the document]
Comment (ldelossa) [reply .luacheckrc:1]: Isnt the suggested place for actually in the LSPs launch settings? 

https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.txt#sumneko_lua

[the remark this answers is not in the document]
Comment (mrjones2014) [reply .luacheckrc:1]: Yes, I have that as well, but I also use `luacheck` and the warnings still come from `luacheck` without this. `luacheck` offers additional and some more detailed diagnostics than just sumneko_lua itself.

[the remark this answers is not in the document]
Comment (ldelossa) [reply .luacheckrc:1]: Gotchya, SGTM.

**Comment by ldelossa** [comment]
Cool.

**Pull request merged**